### ISSUE TITLE
Suggestion: Spawn silverfish as fishing trap

### ISSUE THREAD

**Issue by ghost** [mcMMO-Dev/mcMMO#3573]
I'd like fishing to occasionally spawn a silverfish because it's a trap that can be fought off, rather than instantly hurting the player. This would benefit servers that don't allow AFK fishing because when players are actively fishing they can fight off the silverfish, but when players AFK fish they will eventually be killed by them.

**Comment by Zappa2510** [comment]
well thats what the tnt and potions are for, plus the kraken. most afk farms ive seen you have to spam click and when that happens it summons the kraken and kills them.

**Comment by eightbitx** [comment]
@Zappa2510 I would like _more_ traps, is what I'm saying. Or _better_ traps. Silverfish, being a mob, can be ran away from or strategically fought. Tnt on the other hand just blows up. The kraken is also overkill - loud, annoying, affects regular players with false positives. An occasional trap that would eventually kill the player if left unattended (like silverfish) would be a calmer solution.

**Comment by Zappa2510** [comment]
true...but most ive seen people who try to abuse afk fishing are usually alone and closed up in a small location.

**Comment by Romaq** [comment]
We've not seen the kraken using a tool like AutoClick on a note-block with a half-second pause between clicks. I'm not clear if it's long enough it doesn't trigger the kraken anyway, or why it never spawned. And while I'm the server-op, a Man's Gotta Do What a Man's Gotta Do, and I'm Gonna Do what my wife tells me to... so I had to pull out the traps. :( But if my wife would let me have them in, pulling up a silver-fish as a trap would help make sure people are not AFK without at least an ear out.

**Comment by Zappa2510** [comment]
only time the kraken comes out if you hold down the mouse button and it spams it. normal fishing wont trigger it. if you wait for the fish to get the line then pull it in and recast it wont summon the kraken.

**Comment by nossr50** [comment]
I could probably implement a less death oriented solution to prevent afk fishing

**Comment by nossr50** [comment]
I could probably put something in to alert admins to auto-fishing

**Comment by Romaq** [comment]
FYI per https://www.minecraftforum.net/forums/mapping-and-modding-java-edition/minecraft-mods/1289421-autofish-mod-for-minecraft-1-12-2

Quote: I am a server administrator and would like to disable this mod on my server. How can I do this?
This mod supports ClientPermissions. Follow the instructions and add it to your server to prevent players from using the mod. /Quote

Client Permissions would allow you to forbid this AFK fishing tool (and various other LiteLoader mod client mods supporting Client Permissions) to simply not work on the server. Other mods may be possible, and harder to detect depending on how the mod was coded, but NOT allowing them where possible in the server itself is a good start.

I do support the "catch AFK Fishers with mcMMO", and the server /op sets the server rules others should abide by. But as mentioned before, my wife and I make use of AFK fishing on our server as part of the game. On HermitCraft, we note they also make heavy use of AFK fishing, so the option to allow AFK fishing in the .config if allowed on the server is strongly supported as well.

**Comment by eightbitx** [comment]
> FYI per https://www.minecraftforum.net/forums/mapping-and-modding-java-edition/minecraft-mods/1289421-autofish-mod-for-minecraft-1-12-2
> 
> Quote: I am a server administrator and would like to disable this mod on my server. How can I do this?
> This mod supports ClientPermissions. Follow the instructions and add it to your server to prevent players from using the mod. /Quote
> 
> Client Permissions would allow you to forbid this AFK fishing tool (and various other LiteLoader mod client mods supporting Client Permissions) to simply not work on the server. Other mods may be possible, and harder to detect depending on how the mod was coded, but NOT allowing them where possible in the server itself is a good start.
> 
> I do support the "catch AFK Fishers with mcMMO", and the server /op sets the server rules others should abide by. But as mentioned before, my wife and I make use of AFK fishing on our server as part of the game. On HermitCraft, we note they also make heavy use of AFK fishing, so the option to allow AFK fishing in the .config if allowed on the server is strongly supported as well.

That plugin hasn't been updated since 1.8.

**Comment by nossr50** [comment]
@Romaq If we add in measures to combat AFK fishing they will be configurable (aka you can turn them off or tweak them).
